File: distance-service/src/main/java/pl/edu/pg/benchmarking/distance/service/SphericalDistanceService.java
package pl.edu.pg.benchmarking.distance.service;

import org.springframework.stereotype.Service;

@Service
public class SphericalDistanceService {

    public double calculateDistance(double latitude1, double longitude1, double latitude2, double longitude2) {
        return Math.acos(
                    Math.sin(Math.toRadians(latitude1)) * Math.sin(Math.toRadians(latitude2))
                    + Math.cos(Math.toRadians(latitude1)) * Math.cos(Math.toRadians(latitude2)) * Math.cos(Math.toRadians(longitude1 - longitude2))
                ) * 6371;
    }
}

File: Flask/monolith/website/distance/spherical_distance_service.py
import math
from website.distance.point import Point


class SphericalDistanceService:
    def __init__(self) -> None:
        pass

    def calculate_distance_between_points(self, point1: Point, point2: Point) -> float:
        return self.calculate_distance(point1.latitude, point1.longitude,
            point2.latitude, point2.longitude)

    def calculate_distance(self, latitude1: float, longitude1: float, latitude2: float,
        longitude2: float) -> float:
        return math.acos(math.sin(math.radians(latitude1)) * math.sin(math.radians(latitude2)) + \
            math.cos(math.radians(latitude1)) * math.cos(math.radians(latitude2)) * \
            math.cos(math.radians(longitude1 - longitude2))) * 6371
